sled-agent/src/http_entrypoints.rs

```diff
@@ -706,3 +708,40 @@ async fn write_network_bootstore_config(
 
     Ok(HttpResponseUpdatedNoContent())
 }
+
+/// Add a sled to a rack that was already initialized via RSS
+#[endpoint {
+    method = PUT,
+    path = "/sleds"
+}]
+async fn add_sled_to_initialized_rack(
+    rqctx: RequestContext<SledAgent>,
+    body: TypedBody<AddSledRequest>,
+) -> Result<HttpResponseUpdatedNoContent, HttpError> {
+    let sa = rqctx.context();
+    let request = body.into_inner();
+
+    // Perform some minimal validation
+    if request.start_request.body.use_trust_quorum
+        && !request.start_request.body.is_lrtq_learner
+    {
+        return Err(HttpError::for_bad_request(
+            None,
+            "New sleds must be LRTQ learners if trust quorum is in use"
+                .to_string(),
+        ));
+    }
+
+    crate::sled_agent::add_sled_to_initialized_rack(
+        sa.logger().clone(),
+        request.sled_id,
+        request.start_request,
+    )
+    .await
+    .map_err(|e| {
+        HttpError::for_internal_error(format!(
```

Comment:
Nit / question: `for_internal_error` will only put this error message in the log, and returns an opaque "500 internal error" to the client. Since this is a pretty low-level API anyway, do we want to push a more detailed error out to the client if something goes wrong? We can still use a 500-level code with a more detailed error; it's just slightly more verbose ([example from wicketd](https://github.com/oxidecomputer/omicron/blob/dbf01fddbfc9c9b836c173f70ed80340c9230d09/wicketd/src/http_entrypoints.rs#L449-L456)).

Reply:
Ooh, that's much better! Added in 6552bb8f82c6289130879f1c41b6b7e2960deb51

Reply:
Oops, and I just fixed the error code in f151bb876959bf3bbd8415c1db754ffd267e9b80